theme/views.py

```diff
@@ -499,11 +504,52 @@ def dashboard(request, template="pages/dashboard.html"):
     my_username = request.user.username
     user = User.objects.get(username=my_username)
     my_recent = Variable.recent_resources(user, days=60, n_resources=5)
-
-    context = {'recent': my_recent}
+    today = date.today()
+    beginning = today - timedelta(30)
+
+    my_recommended_resources_list = RecommendedResource.objects\
+        .filter(state__lte=Status.STATUS_EXPLORED, user__username=my_username)\
+        .order_by('-relevance')[:Status.RECOMMENDATION_LIMIT]
+    resources_matched_keywords = set()
+    # mark relevant records as shown
+    my_recommended_resources_context_list = []
+    for r in my_recommended_resources_list:
+        if r.state == Status.STATUS_NEW:
+            r.shown()
+        if r.rec_type == 'Propensity' and r.relevance > 0:
+            recommended_resource = r.candidate_resource
+            # keywords = r.keywords.all()
+            common_keywords = []
+           
+            if len(r.keywords) > 0:
+                for k, v in list(r.keywords.items()):
+                    resources_matched_keywords.add(k)
+                    common_keywords.append(k)
```

Comment:
aren't these lists the same, except one is a set and the other isn't?  Keywords should be a set anyways, so they should always be equal sets.

Reply:
nvm, I see that one is for all resources and common is only for the one